File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.cpp
class Solution 
{
public:
    vector <int> kWeakestRows(vector<vector<int>>& mat, int k) 
    {
        vector < pair < int , int > > v;
        for (auto &row : mat)
        {
            int l = 0 , r = row.size() - 1 , ans = 0;
            while (l <= r)
            {
                int mid = (l + r) >> 1;
                if (row[mid] == 1)
                    l = mid + 1 , ans = mid + 1;
                else 
                    r = mid - 1;
            }
            v.push_back({ans , v.size()});
        }
        sort(v.begin() , v.end());
        while(v.size() > k)
            v.pop_back();
        vector < int > ans;
        for (auto &it : v)
            ans.push_back(it.second);
        return ans;      
    }
};

File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.js
/**
 * @param {number[][]} mat
 * @param {number} k
 * @return {number[]}
 */
var kWeakestRows = function(mat, k) {
    let v = [];
    for (let i = 0; i < mat.length; i++) {
    let row = mat[i];
    let l = 0, r = row.length - 1, ans = 0;
    while (l <= r) {
      let mid = Math.floor((l + r) / 2);
      if (row[mid] === 1) {
           l = mid + 1;
           ans = mid + 1;
         } else {
           r = mid - 1;
         }
       }
       v.push([ans, i]);
     }
      v.sort((a, b) => a[0] - b[0]);
      while (v.length > k) {
        v.pop();
      }
      let ans = [];
      for (let i = 0; i < v.length; i++) {
        ans.push(v[i][1]);
      }
      return ans;
};